swupd/bundleinfo.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package swupd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type bundleInfo struct {
```

Comment:
This looks nearly exactly like the `bundle` type. Why do we have both?

Reply:
Because I didn't want to import the builder package just to get the bundle struct... I suppose I could move both definitions to helpers? or some internal package not yet defined?

Reply:
That's a pretty good reason. And [The Rule of Three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) argues we shouldn't refactor it yet.

[The Go Blog](https://blog.golang.org/package-names) argues against us having a `helper` package altogether, and for pretty good reason.

Probably (someday) the right thing to do is to move the purely "bundle" related types and methods (the stuff that currently lives in `bundleset.go`, mostly) into a non-exported/internal `bundle` package that both builder and swupd import.

As I've said before, Mixer could use an overall organizational code refactor.